cocotb/decorators.py

```diff
@@ -567,6 +570,8 @@ async def f(*args, **kwargs):
         self.timeout_unit = timeout_unit
         self.expect_fail = expect_fail
         if expect_error is True:
+            warnings.warn(
+                "`except_error=True` option to `cocotb.test` has been deprecated, specify an expected Python Exception instead")
             expect_error = (Exception,)
         elif expect_error is False:
             expect_error = ()
```

Comment:
Should we deprecate `False` too?

Reply:
It's currently used for spelling `expect_error=Exception if cond else False`. We could deprecate `False`, and replace it with `None`? So instead of `Union[bool, Exception]` it's `Optional[Exception]`, which is a more common interface type.

Reply:
We already allow `Exception if cond else ()`, which is shorter than `False`. `None` would be ok, but given the choice between `Union[Exception, Tuple[Exception], None]` and `Union[Exception, Tuple[Exception]]`, I'd lean towards the shorter choice.

Reply:
I replaced all instances of `expect_error=False` and updated the default value and docstring.